languagetool-standalone/src/main/java/org/languagetool/gui/PopUpRemoveLanguage.java

```diff
@@ -0,0 +1,78 @@
+package org.languagetool.gui;
+
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.util.ResourceBundle;
+
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.JPopupMenu;
+
+import org.languagetool.JLanguageTool;
+
+public class PopUpRemoveLanguage extends JPopupMenu {
+
+  private final ResourceBundle messages;
+  private final JMenuItem removeOption;
+  private final LanguageAdapter languageAdapter;
+
+  public PopUpRemoveLanguage() {
+    languageAdapter = (LanguageAdapter) LanguageComboBoxModel.languageComboBoxModel.getSelectedItem();
+    messages = JLanguageTool.getMessageBundle(languageAdapter.getLanguage());
+    removeOption = new JMenuItem(messages.getString("guiRemoveOptionText"));
+    add(removeOption);
+    removeOption.addMouseListener(new OnClickListener());
+  }
+}
+
+class OnClickListener implements MouseListener {
+  @Override
+  public void mousePressed(MouseEvent e) {
+    LanguageComboBoxModel comboBoxModel = LanguageComboBoxModel.languageComboBoxModel;
```

Comment:
ComboBoxModel will hold all the settings.So,the set of languages are taken from here and are removed from here only.If you want all the languages that are deleted,we can just find set A - set B where set A=the original list which is unmodifiable and which we get at Languages.get() and set B is the set coming from  by iterating the elements in comboBox [model.getElementAt(model.getSize())].